src/transport/datamodel/utransport.rs

```diff
@@ -11,19 +11,61 @@
  * SPDX-License-Identifier: Apache-2.0
  ********************************************************************************/
 
-use crate::transport::datamodel::{UAttributes, UListener, UPayload, UStatus};
-use crate::uri::datamodel::{UEntity, UUri};
+use async_trait::async_trait;
+use std::sync::mpsc::Sender;
 
+use crate::uprotocol::{UAttributes, UEntity, UMessage, UPayload, UStatus, UUri};
+
+/// `UTransport` is the uP-L1 interface that provides a common API for uE developers to send and receive messages.
+///
+/// Implementations of `UTransport` contain the details for connecting to the underlying transport technology and
+/// sending `UMessage` using the configured technology. For more information, please refer to
+/// [uProtocol Specification](https://github.com/eclipse-uprotocol/uprotocol-spec/blob/main/up-l1/README.adoc).
+#[async_trait]
 pub trait UTransport {
-    /// API to register the calling uE with the underlying transport implementation.
-    fn register(&self, uentity: UEntity, token: &[u8]) -> UStatus;
```

Comment:
What is this register API?

Reply:
That was what was in the Java SDK months ago... but gone now, replaced by authenticate